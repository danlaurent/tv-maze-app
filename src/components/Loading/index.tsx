import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
    marginBottom: 32,
  },
});

const Loading = ({ testID }: { testID?: string }) => (
  <View testID={testID} style={styles.container}>
    <ActivityIndicator size='large' color='white' />
  </View>
);

export default Loading;
