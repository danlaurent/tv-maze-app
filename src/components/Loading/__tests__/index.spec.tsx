import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import Loading from '..';

describe('Loading', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(<Loading />);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
