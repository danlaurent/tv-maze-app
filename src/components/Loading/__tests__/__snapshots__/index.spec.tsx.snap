// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 32,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
